core/src/main/java/org/infinispan/util/rxjava/FlowableFromIntSetFunction.java

```diff
@@ -0,0 +1,424 @@
+
+            if (cancelled) {
+               return;
+            }
+
+            boolean b;
+
+            try {
+               b = it.hasNext();
+            } catch (Throwable ex) {
+               Exceptions.throwIfFatal(ex);
+               a.onError(ex);
+               return;
+            }
+
+
+            if (!b) {
+               if (!cancelled) {
+                  a.onComplete();
+               }
+               return;
+            }
+         }
+      }
+
+      @Override
+      void slowPath(long r) {
+         long e = 0L;
+         PrimitiveIterator.OfInt it = this.it;
+         IntFunction<T> intFunction = this.intFunction;
+         Subscriber<? super T> a = actual;
+
+         for (;;) {
+
+            while (e != r) {
+
+               if (cancelled) {
+                  return;
+               }
+
+               T t;
+
+               try {
+                  t = intFunction.apply(it.nextInt());
+               } catch (Throwable ex) {
+                  Exceptions.throwIfFatal(ex);
+                  a.onError(ex);
+                  return;
+               }
+
+               if (cancelled) {
+                  return;
+               }
+
+               if (t == null) {
+                  a.onError(new NullPointerException("IntFunction.apply(OfInt.nextInt()) returned a null value"));
+                  return;
+               } else {
+                  a.onNext(t);
+               }
+
+               if (cancelled) {
+                  return;
+               }
+
+               boolean b;
+
+               try {
+                  b = it.hasNext();
+               } catch (Throwable ex) {
+                  Exceptions.throwIfFatal(ex);
+                  a.onError(ex);
+                  return;
+               }
+
+               if (!b) {
+                  if (!cancelled) {
+                     a.onComplete();
+                  }
+                  return;
+               }
+
+               e++;
+            }
+
+            r = get();
+            if (e == r) {
+               r = addAndGet(-e);
+               if (r == 0L) {
+                  return;
+               }
+               e = 0L;
+            }
+         }
+      }
+
+   }
+
+   static final class IteratorConditionalSubscription<T> extends BaseRangeSubscription<T> {
+
+
+      private static final long serialVersionUID = -6022804456014692607L;
+
+      final ConditionalSubscriber<? super T> actual;
+
+      IteratorConditionalSubscription(ConditionalSubscriber<? super T> actual, PrimitiveIterator.OfInt it,
+            IntFunction<T> intFunction) {
+         super(it, intFunction);
+         this.actual = actual;
+      }
+
+      @Override
+      void fastPath() {
+         PrimitiveIterator.OfInt it = this.it;
+         IntFunction<T> intFunction = this.intFunction;
+         ConditionalSubscriber<? super T> a = actual;
+         for (;;) {
+            if (cancelled) {
+               return;
+            }
+
+            T t;
```

Comment:
Nitpick, but all of the whitespace in `fastPath` and `slowPath` seems excessive

Reply:
This is because this is taken as an example from the Iterator Flowable class in rxjava2. Let me add that to the docs of this class.